File: backend/app/api/einstellungen.py
"""Einstellungs-Endpunkt — zeigt aktuelle (nicht-geheime) Konfiguration."""

from __future__ import annotations

from fastapi import APIRouter

from app.config import einstellungen

router = APIRouter()


@router.get("")
async def lies() -> dict:
    """Gibt die aktive Konfiguration zurueck — Secrets nur als Boolean-Flags."""
    return {
        "llm": {
            "provider": einstellungen.llm_provider,
            "anthropic_modell": einstellungen.anthropic_modell,
            "anthropic_api_key_gesetzt": bool(einstellungen.anthropic_api_key),
            "openai_modell": einstellungen.openai_modell,
            "openai_basis_url": einstellungen.openai_basis_url or None,
            "openai_api_key_gesetzt": bool(einstellungen.openai_api_key),
        },
        "simulation": {
            "max_agenten": einstellungen.max_agenten,
            "simulations_schritte": einstellungen.simulations_schritte,
            "demo_daten_einspielen": einstellungen.demo_daten_einspielen,
        },
        "datenbank_dialekt": einstellungen.datenbank_url.split("+")[0].split(":")[0],
    }

File: frontend/src/api/einstellungen.ts
import { apiClient } from '@/api/client';

export interface Einstellungen {
  llm: {
    provider: string;
    anthropic_modell: string;
    anthropic_api_key_gesetzt: boolean;
    openai_modell: string;
    openai_basis_url: string | null;
    openai_api_key_gesetzt: boolean;
  };
  simulation: {
    max_agenten: number;
    simulations_schritte: number;
    demo_daten_einspielen: boolean;
  };
  datenbank_dialekt: string;
}

export const einstellungenApi = {
  hole: () => apiClient.get<Einstellungen>('/api/einstellungen').then((r) => r.data),
};
